code/game/objects/structures/window.dm

```diff
@@ -108,22 +108,24 @@ var/global/wcCommon = pick(list("#379963", "#0d8395", "#58b5c3", "#49e46e", "#8f
 
 	return 1
 
-/obj/structure/window/hitby(atom/movable/AM)
-	if(!CanPass(AM, get_step(src, AM.dir))) //So thrown objects that cross a tile with non-full windows will no longer hit the window even if it isn't visually obstructing the path.

```

Comment:
Yeah, that other PR was just a patchwork fix. This PR fixes the entirety of the problem